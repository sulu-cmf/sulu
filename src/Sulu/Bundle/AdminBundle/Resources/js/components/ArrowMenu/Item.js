// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import itemStyles from './item.scss';

type Props = {
    value: *,
    children: string,
    active: boolean,
    icon?: string,
    onClick?: (value: *) => void,
};

export default class Item extends React.PureComponent<Props> {
    handleButtonClick = () => {
        const {onClick, value} = this.props;

        if (!onClick) {
            return;
        }

        onClick(value);
    };

    static defaultProps = {
        active: false,
    };

    render() {
        const {
            children,
            active,
            icon,
        } = this.props;

        const itemClass = classNames(
            itemStyles.item,
            {
                [itemStyles.active]: active,
            }
        );

        return (
            <div className={itemClass} onClick={this.handleButtonClick}>
                <span className={itemStyles.icon}>
                    {icon && active && <Icon className={itemStyles.icon} name={icon} />}
                </span>
                <span>
                    {children}
                </span>
            </div>
        );
    }
}
