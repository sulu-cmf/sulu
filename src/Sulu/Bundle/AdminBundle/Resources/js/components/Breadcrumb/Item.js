// @flow
import React from 'react';
import itemStyles from './item.scss';

type Props = {
    onClick?: (value?: string | number) => void,
    children: string,
    value?: string | number,
};

export default class Item extends React.PureComponent<Props> {
    handleClick = () => {
        const {
            value,
            onClick,
        } = this.props;

        if (onClick) {
            onClick(value);
        }
    };

    render() {
        const {
            onClick,
            children,
        } = this.props;

        return (
            <button
                className={itemStyles.item}
                disabled={!onClick}
                onClick={this.handleClick}
            >
                {children}
            </button>
        );
    }
}
