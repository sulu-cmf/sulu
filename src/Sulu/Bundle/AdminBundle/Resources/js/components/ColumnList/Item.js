// @flow
import React from 'react';
import classNames from 'classnames';
import CroppedText from '../CroppedText';
import Icon from '../Icon';
import ItemButton from './ItemButton';
import type {ItemButtonConfig} from './types';
import itemStyles from './item.scss';

type Props = {|
    active: boolean,
    buttons?: Array<ItemButtonConfig>,
    children: string,
    disabled: boolean,
    hasChildren: boolean,
    id: string | number,
    onClick?: (id: string | number) => void,
    selected: boolean,
|};

export default class Item extends React.Component<Props> {
    static defaultProps = {
        active: false,
        disabled: false,
        hasChildren: false,
        selected: false,
    };

    handleClick = () => {
        if (this.props.onClick) {
            this.props.onClick(this.props.id);
        }
    };

    renderButtons = () => {
        const {buttons, id} = this.props;

        if (!buttons) {
            return null;
        }

        return buttons.map((button: ItemButtonConfig, index: number) => {
            const key = `button-${index}`;

            return (
                <ItemButton id={id} key={key} config={button} />
            );
        });
    };

    render() {
        const {active, children, disabled, hasChildren, selected} = this.props;

        const itemClass = classNames(
            itemStyles.item,
            {
                [itemStyles.active]: active,
                [itemStyles.disabled]: disabled,
                [itemStyles.selected]: selected,
            }
        );

        return (
            <div onClick={this.handleClick} className={itemClass}>
                <span className={itemStyles.buttons}>
                    {this.renderButtons()}
                </span>
                <span className={itemStyles.text}>
                    <CroppedText>{children}</CroppedText>
                </span>
                {hasChildren &&
                    <Icon className={itemStyles.children} name="su-angle-right" />
                }
            </div>
        );
    }
}
