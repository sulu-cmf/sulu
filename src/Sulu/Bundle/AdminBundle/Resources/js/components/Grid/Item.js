// @flow
import React from 'react';
import type {Element} from 'react';
import classNames from 'classnames';
import type {BaseItemProps} from './types';
import BaseItem from './BaseItem';
import itemStyles from './item.scss';

type Props = BaseItemProps & {
    children?: Element<*>,
    className?: string,
};

export default class Item extends React.PureComponent<Props> {
    static defaultProps = {
        size: 12,
        spaceAfter: 0,
        spaceBefore: 0,
    };

    render() {
        const {
            children,
            className,
            ...others
        } = this.props;

        const itemClass = classNames([
            itemStyles.item,
            className,
        ]);

        return (
            <BaseItem {...others} className={itemClass}>
                {children}
            </BaseItem>
        );
    }
}
