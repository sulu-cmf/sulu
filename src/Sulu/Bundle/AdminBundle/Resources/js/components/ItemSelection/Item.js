// @flow
import React from 'react';
import type {Node} from 'react';
import Icon from '../Icon';
import itemStyles from './item.scss';

const DRAG_ICON = 'ellipsis-v';
const REMOVE_ICON = 'times';

type Props = {
    id: string | number,
    children: Node,
    onRemove: (id: string | number) => void,
};

export default class Item extends React.PureComponent<Props> {
    handleRemove = () => {
        this.props.onRemove(this.props.id);
    };

    render() {
        const {children} = this.props;

        return (
            <div className={itemStyles.item}>
                <button className={itemStyles.dragButton}>
                    <Icon name={DRAG_ICON} />
                </button>
                <div className={itemStyles.content}>
                    {children}
                </div>
                <button
                    className={itemStyles.removeButton}
                    onClick={this.handleRemove}
                >
                    <Icon name={REMOVE_ICON} />
                </button>
            </div>
        );
    }
}
