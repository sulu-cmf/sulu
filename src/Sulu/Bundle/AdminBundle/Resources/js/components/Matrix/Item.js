// @flow
import React from 'react';
import classNames from 'classnames';
import Icon from '../Icon';
import matrixStyles from './matrix.scss';

type Props = {|
    disabled: boolean,
    icon: string,
    name: string,
    onChange?: (name: string, value: boolean) => void,
    title?: string,
    value: boolean,
|};

export default class Item extends React.PureComponent<Props> {
    static defaultProps = {
        disabled: false,
        value: false,
    };

    handleClick = () => {
        const {
            name,
            onChange,
            value,
        } = this.props;

        if (!onChange) {
            return;
        }

        onChange(name, !value);
    };

    render() {
        const {
            disabled,
            icon,
            name,
            title,
            value,
        } = this.props;
        const itemClass = classNames(
            matrixStyles.item,
            {
                [matrixStyles.selected]: value,
                [matrixStyles.disabled]: disabled,
            }
        );

        const itemTitle = title ? title : name.charAt(0).toUpperCase() + name.slice(1);

        return (
            <div className={itemClass} onClick={!disabled ? this.handleClick : undefined} title={itemTitle}>
                <Icon name={icon} />
            </div>
        );
    }
}
