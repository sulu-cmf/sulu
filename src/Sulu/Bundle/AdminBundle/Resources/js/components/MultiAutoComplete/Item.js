// @flow
import React from 'react';
import Icon from '../../components/Icon';
import itemStyles from './item.scss';

type Props = {
    children: string,
    onDelete: (value: Object) => void,
    value: Object,
};

export default class Item extends React.Component<Props> {
    handleDelete = () => {
        const {onDelete, value} = this.props;
        onDelete(value);
    };

    render() {
        const {children} = this.props;

        return (
            <div className={itemStyles.item}>
                {children}
                <Icon className={itemStyles.icon} name="su-times" onClick={this.handleDelete} />
            </div>
        );
    }
}
