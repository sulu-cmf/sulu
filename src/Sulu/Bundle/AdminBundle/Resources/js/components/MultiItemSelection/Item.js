// @flow
import React from 'react';
import type {Node} from 'react';
import {SortableHandle} from 'react-sortable-hoc';
import Icon from '../Icon';
import itemStyles from './item.scss';

const DRAG_ICON = 'su-more';
const REMOVE_ICON = 'su-trash-alt';

type Props = {
    id: string | number,
    index: number,
    children: Node,
    onRemove?: (id: string | number) => void,
};

export default class Item extends React.PureComponent<Props> {
    createDragHandle() {
        return SortableHandle(({children, className}) => (
            <span className={className}>{children}</span>
        ));
    }

    handleRemove = () => {
        if (this.props.onRemove) {
            this.props.onRemove(this.props.id);
        }
    };

    render() {
        const {
            index,
            onRemove,
            children,
        } = this.props;
        const DragHandle = this.createDragHandle();

        return (
            <div className={itemStyles.item}>
                <DragHandle className={itemStyles.dragHandle}>
                    <Icon name={DRAG_ICON} />
                    <span className={itemStyles.index}>{index}</span>
                </DragHandle>
                <div className={itemStyles.content}>
                    {children}
                </div>
                {onRemove &&
                    <button
                        type="button"
                        className={itemStyles.removeButton}
                        onClick={this.handleRemove}
                    >
                        <Icon name={REMOVE_ICON} />
                    </button>
                }
            </div>
        );
    }
}
