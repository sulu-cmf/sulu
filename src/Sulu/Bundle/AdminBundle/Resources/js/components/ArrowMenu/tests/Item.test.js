// @flow
import React from 'react';
import {render, mount} from 'enzyme';
import Item from '../Item';

test('Render default Item', () => {
    expect(render(<Item value="test">Test Item</Item>)).toMatchSnapshot();
});

test('Render active Item', () => {
    expect(render(<Item active={true} icon="fa-home" value="house">My House</Item>)).toMatchSnapshot();
});

test('Clicking the left and right button inside the header should call the right handler', () => {
    const clickHandler = jest.fn();

    const item = mount(
        <Item active={true} icon="fa-home" onClick={clickHandler} value="house">My House</Item>
    );

    item.simulate('click');
    expect(clickHandler).toBeCalledWith('house');
});
