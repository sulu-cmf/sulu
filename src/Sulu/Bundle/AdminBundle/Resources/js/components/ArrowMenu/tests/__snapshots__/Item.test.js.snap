// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render active Item 1`] = `
<div
  class="item active"
>
  <span
    class="icon"
  >
    <span
      aria-label="fa-home"
      class="fa fa-home icon"
    />
  </span>
  <span>
    My House
  </span>
</div>
`;

exports[`Render default Item 1`] = `
<div
  class="item"
>
  <span
    class="icon"
  />
  <span>
    Test Item
  </span>
</div>
`;
